.github/workflows/lint.yaml

```diff
@@ -0,0 +1,52 @@
+name: Lint and Format
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - 'master'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+
+    - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+    - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-lint-${{ hashFiles('**/poetry.lock') }}
+
+    - name: Install dependencies
+      if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+      run: poetry install --no-interaction --no-root
+
+    - name: Check formatting
+      run: |
+        poetry run yapf -r -p --diff
+        poetry run isort --check-only
```

Comment:
I don't know if this will actually work because I'm not sure that yapf and isort are guaranteed to be compatible?

Reply:
yapf doesn't sort imports, as far as I know

Reply:
It does do line wrapping though. I seem to recall a past experience of mine where the two formatters disagreed.